snap-geometry-edits/src/main/java/com/esri/arcgismaps/sample/snapgeometryedits/screens/MainScreen.kt

```diff
@@ -0,0 +1,192 @@
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.unit.dp
+import com.arcgismaps.geometry.GeometryType
+import com.arcgismaps.mapping.view.MapViewInteractionOptions
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.esri.arcgismaps.sample.sampleslib.components.BottomSheet
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+import com.esri.arcgismaps.sample.snapgeometryedits.R
+import com.esri.arcgismaps.sample.snapgeometryedits.components.MapViewModel
+import com.esri.arcgismaps.sample.snapgeometryedits.components.SnapSettings
+
+/**
+ * Main screen layout for the sample app
+ */
+@Composable
+fun MainScreen(sampleName: String) {
+    // coroutineScope that will be cancelled when this call leaves the composition
+    val sampleCoroutineScope = rememberCoroutineScope()
+    // get the application property that will be used to construct MapViewModel
+    val sampleApplication = LocalContext.current.applicationContext as Application
+    // create a ViewModel to handle MapView interactions
+    val mapViewModel = remember { MapViewModel(sampleApplication, sampleCoroutineScope) }
+    // the collection of graphics overlays used by the MapView
+    val graphicsOverlayCollection = listOf(mapViewModel.graphicsOverlay)
+
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) },
+        content = {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it)
+            ) {
+                MapView(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .weight(1f),
+                    arcGISMap = mapViewModel.map,
+                    geometryEditor = mapViewModel.geometryEditor,
+                    graphicsOverlays = graphicsOverlayCollection,
+                    mapViewProxy = mapViewModel.mapViewProxy,
+                    mapViewInteractionOptions = MapViewInteractionOptions(isMagnifierEnabled = true),
+                    onSingleTapConfirmed = mapViewModel::identify,
+                    onPan = { mapViewModel.dismissBottomSheet() }
+                )
+                Row(
+                    modifier = Modifier
+                        .padding(12.dp)
+                        .fillMaxWidth(),
+                ) {
```

Comment:
This `Row` can be extrapolated into a Composable function outside of the `MainScreen` function.  Maybe something along the lines of: `GeometryOptionsMenu(mapViewModel: MapViewModel)` which consumes the ViewModel.

Reply:
I tried this approach and will need to look at it further because the buttons now become enabled before the layers are loaded when moving the row to an outside composable function.

Reply:
I was able solve this by ensuring the map is loaded before loading the operational layers. Thanks @fpearson1111 for the help.